### PR TITLE
[PRD-6159] - upgrading the apache poi libraries

### PR DESCRIPTION
**Pull request by lsaikrishna89**
…iter - fix

### CODE REVIEW

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :warning: Build finished in 27m 24s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl \
assemblies/client,engine,plugins/excel/core,plugins/pentaho-reporting/assemblies/plugin,plugins/xml
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 1 test(s) failed:

<details><summary><b>org.pentaho.di.trans.steps.excelwriter.ExcelWriterFullTest.testDataValidationExtension</b> (click to expand)</summary><p>

```
validation not extended
```

</p></details>

  
Tests run: **2442**, Failures: **1**, Skipped: **13** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9443/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: plugins/excel/core/src/main/java/org/pentaho/di/trans/steps/excelwriter/ExcelWriterStep.java [hunk 1]
```diff
@@ -687,10 +720,10 @@ public void prepareNextOutputFile() throws KettleException {
...
       try ( InputStream inputStream = KettleVFS.getInputStream( data.file ) ) {
         if ( XLSX.equalsIgnoreCase( meta.getExtension() ) ) {
-          // Ignore, by now, if it's to use streaming!
-          // In that case, one needs to initialize it later, after writing header/template, because
-          // SXSSFWorkbook can't read/rewrite existing data, only append.
-          data.wb = new XSSFWorkbook( inputStream );
+          Workbook workbook = WorkbookFactory.create(inputStream);
+          SXSSFWorkbook sxssfWorkbook = new SXSSFWorkbook();
+          copyData(workbook, sxssfWorkbook);
+          data.wb = sxssfWorkbook;
```
Comment (tgf) [inline plugins/excel/core/src/main/java/org/pentaho/di/trans/steps/excelwriter/ExcelWriterStep.java:726]: This is selecting streaming mode even if `meta.isStreamingData()` (`Stream XLSX Data` option) is set to false.
If you want to remove that option, you need to also remove it from the UI. This shouldn't be done in a bug, and you should seek PM approval first.

**Review by tgf — CHANGES_REQUESTED** [review]

File: plugins/excel/core/src/main/java/org/pentaho/di/trans/steps/excelwriter/ExcelWriterStep.java (hunk 1 above)
Comment (lsaikrishna89) [reply plugins/excel/core/src/main/java/org/pentaho/di/trans/steps/excelwriter/ExcelWriterStep.java]: I dont want to remove that option, just updated the code am checking the meta.isStreamingData condition if we want to create the SXSSFWorkbook instance

Comment (tgf) [reply plugins/excel/core/src/main/java/org/pentaho/di/trans/steps/excelwriter/ExcelWriterStep.java]: So now what is the improvement that these changes bring, and how do they fix BACKLOG-41001?

Comment (tgf) [reply plugins/excel/core/src/main/java/org/pentaho/di/trans/steps/excelwriter/ExcelWriterStep.java]: BTW I made a comment in the jira ticket, don't know if it was noticed

Comment (lsaikrishna89) [reply plugins/excel/core/src/main/java/org/pentaho/di/trans/steps/excelwriter/ExcelWriterStep.java]: previously by default we use to create SXSSFWorkbook, when we are dealing with huge amount of data we used to get outofmemory exception, so from now we with the following check meta.isStreamingData(), we will be avoid that by creating the XSSFWorkbook 
If we are crossing this limit  (1,048,575 rows and 16,384 columns), we need to split the excel sheet by rows .

Comment (tgf) [reply plugins/excel/core/src/main/java/org/pentaho/di/trans/steps/excelwriter/ExcelWriterStep.java]: When in streaming mode there wouldn't be an OOM error before. And with it disabled there will still be one after this change.

Comment (lsaikrishna89) [reply plugins/excel/core/src/main/java/org/pentaho/di/trans/steps/excelwriter/ExcelWriterStep.java]: coding wise I am not disabling the streaming mode option, with the steaming mode condition I am just creating the SXSSFWorkbook instance which we were not doing before , with the existing code I have seen like XSSFWorkbook is being created by default which is causing OOM exception

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 35m 34s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl \
assemblies/client,engine,plugins/excel/core,plugins/pentaho-reporting/assemblies/plugin,plugins/xml
```
---
:ok_hand: All tests passed!
 
Tests run: **2442**, Failures: **0**, Skipped: **13** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9443/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 1m 26s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl \
assemblies/client,engine,plugins/excel/core,plugins/pentaho-reporting/assemblies/plugin,plugins/xml
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

File: plugins/excel/core/src/main/java/org/pentaho/di/trans/steps/excelwriter/ExcelWriterStep.java (hunk 1 above)
Comment (tgf) [reply plugins/excel/core/src/main/java/org/pentaho/di/trans/steps/excelwriter/ExcelWriterStep.java]: > I am just creating the SXSSFWorkbook instance which we were not doing before

We were already doing that a bit below:
https://github.com/pentaho/pentaho-kettle/blob/200f687aa127113aad7a49f2dbab97ae5172459a/plugins/excel/core/src/main/java/org/pentaho/di/trans/steps/excelwriter/ExcelWriterStep.java#L752

If you test current 10.2 with the streaming option enabled, there will be no OoM error.

Comment (tgf) [reply plugins/excel/core/src/main/java/org/pentaho/di/trans/steps/excelwriter/ExcelWriterStep.java]: I believe the issue should be reevaluated, taking the streaming option into account.

Comment (lsaikrishna89) [reply plugins/excel/core/src/main/java/org/pentaho/di/trans/steps/excelwriter/ExcelWriterStep.java]: yeap its working fine with streaming option

**Comment by buildguy** [comment]
### :x: Build failed in 6s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 3s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 3s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 3s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 1m 19s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl \
assemblies/client,engine,plugins/excel/core,plugins/pentaho-reporting/assemblies/plugin,plugins/xml
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 29m 12s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl \
assemblies/client,engine,plugins,plugins/excel/core,plugins/pentaho-reporting/assemblies/plugin,plugins/xml
```
---
:ok_hand: All tests passed!
 
Tests run: **2442**, Failures: **0**, Skipped: **13** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9443/10/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 13m 33s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl \
assemblies/client,engine,plugins,plugins/excel/core,plugins/pentaho-reporting/assemblies/plugin,plugins/xml
```
---
:ok_hand: All tests passed!
 
Tests run: **2442**, Failures: **0**, Skipped: **13** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9443/11/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: plugins/excel/core/src/main/java/org/pentaho/di/trans/steps/excelwriter/ExcelWriterStep.java (hunk 1 above)
Comment (lsaikrishna89) [reply plugins/excel/core/src/main/java/org/pentaho/di/trans/steps/excelwriter/ExcelWriterStep.java]: Hi, we have rolledback changes related to streaming version workbook, going with changes related to upgrading the apache poi

File: engine/pom.xml [hunk 2]
```diff
@@ -51,7 +51,7 @@
     <odfdom-java.version>0.8.6</odfdom-java.version>
     <commons-collections4.version>4.4</commons-collections4.version>
     <commons-math3.version>3.6.1</commons-math3.version>
-    <xmlbeans.version>3.1.0</xmlbeans.version>
+    <xmlbeans.version>5.2.0</xmlbeans.version>
```
Comment (peterrinehart) [inline engine/pom.xml:54]: Is this needed for the POI library update?

**Review by peterrinehart — CHANGES_REQUESTED** [review]

File: engine/pom.xml (hunk 2 above)
Comment (lsaikrishna89) [reply engine/pom.xml:54]: yes @peterrinehart , xmlbeans: The Apache XMLBeans library, used by POI to process XML-based data in OOXML documents.

**Review by peterrinehart — APPROVED** [review]

File: plugins/excel/core/src/main/java/org/pentaho/di/trans/steps/excelwriter/ExcelWriterStep.java (hunk 1 above)
Comment (tgf) [reply plugins/excel/core/src/main/java/org/pentaho/di/trans/steps/excelwriter/ExcelWriterStep.java]: The commit msg still mentions the other issue, can you change that please?

**Pull request merged**